__tests__/integration.test.js

```diff
@@ -1,112 +1,175 @@
-  })
-})
+  });
+
+  it("GET 200: Return an article to the user", async () => {
+    const res = await supertest(app).get("/api/articles/1");
+
+    const article = res.body.article;
+    expect(res.statusCode).toBe(200);
+
+    expect(article.article_id).toBe(1);
+    expect(article).toEqual(
+      expect.objectContaining({
+        author: expect.any(String),
+        title: expect.any(String),
+        body: expect.any(String),
+        topic: expect.any(String),
+        created_at: expect.any(String),
+        votes: expect.any(Number),
+        article_img_url: expect.any(String),
+      }),
+    );
+  });
+
+  it("GET 404: Return an error to the user when an article is not found", async () => {
+    const res = await supertest(app).get("/api/articles/12000");
+
+    expect(res.statusCode).toBe(404);
+    expect(res.body.msg).toBe("Article ID not found");
+  });
+
+  it("GET 400: Return an error to the user when invalid article ID is given", async () => {
+    const res = await supertest(app).get("/api/articles/asdas");
+
+    expect(res.statusCode).toBe(400);
+    expect(res.body.msg).toBe("Invalid input");
+  });
+});
+
+describe("comments", () => {
+  it("POST 201: Should create a new comment for an article", async () => {
+    const res = await supertest(app).post("/api/articles/1/comments").send({
+      username: "butter_bridge",
+      body: "test body",
+    });
+
+    const comment = res.body.comment
+
+    expect(res.statusCode).toBe(201);
+    expect(comment.author).toBe("butter_bridge")
+    expect(comment.body).toBe("test body")
+    expect(comment.article_id).toBe(1)
+  });
+
+  it("POST 400: Should return an error when user gives invalid input for article ID", async () => {
+    const res = await supertest(app).post("/api/articles/asdsa/comments").send({
+      username: "butter_bridge",
+      body: "test body",
+    });
+
+    expect(res.statusCode).toBe(400);
+    expect(res.body.msg).toBe("Invalid input");
+  });
+
+  it("POST 404: Should return an error article with ID does not exist", async () => {
+    const res = await supertest(app).post("/api/articles/12312312/comments").send({
+      username: "butter_bridge",
+      body: "test body",
+    });
+
+    expect(res.statusCode).toBe(404);
+    expect(res.body.msg).toBe("Article ID not found");
+  });
+
+  it("POST 400: Should return an error when user can not be found", async () => {
+    const res = await supertest(app).post("/api/articles/1/comments").send({
+      username: "asd",
+      body: "test body",
+    });
+
+    expect(res.statusCode).toBe(400);
+    expect(res.body.msg).toBe("Bad request");
+  });
+
+  it("POST 401: Should return an error when body is empty", async () => {
+    const res = await supertest(app).post("/api/articles/1/comments").send({
+      username: "butter_bridge",
+      body: "",
+    });
+
+    expect(res.statusCode).toBe(400);
+    expect(res.body.msg).toBe("Body can not be empty");
+  });

```

Comment:
These are all really good error handling tests, but another test to include would be if the given object when making a post request simply did not have one of the necessary properties e.g.

`
{
      body: "test body",
}
`

What should be the expected response here?